flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/connector/kafka/source/enumerator/KafkaEnumeratorTest.java

```diff
@@ -347,6 +347,41 @@ public void testSnapshotState() throws Throwable {
                 state2.assignedPartitions());
     }
 
+    @Test
+    public void testPartitionChangeChecking() throws Throwable {
+        final MockSplitEnumeratorContext<KafkaPartitionSplit> context =
+                new MockSplitEnumeratorContext<>(NUM_SUBTASKS);
+        final KafkaSourceEnumerator enumerator =
```

Comment:
It looks that there is a resource leak here. Supposedly, the `KafkaSourceEnumerator` should be closed after the test. As you may notice the other test method use the try block to do this. However, the `MockSplitEnumeratorContext#close()` did not shutdown the `mainExecutor` and `workerExecutor` correctly. This is an existing bug. Would be good to fix it either in this patch or a followup patch. In either way, we need to close the `KafkaSourceEnumerator` here in the unit tests.